include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h

```diff
@@ -235,3 +235,9 @@ static const char* const kOrtSessionOptionsOptimizedModelExternalInitializersFil
 // Use this config to control the minimum size of the initializer when externalizing it during serialization
 static const char* const kOrtSessionOptionsOptimizedModelExternalInitializersMinSizeInBytes =
     "session.optimized_model_external_initializers_min_size_in_bytes";
+
+// Gemm fastmath mode provides fp32 gemm acceleration with bfloat16 based matmul.
+// Option values:
+// - "0": Gemm FastMath mode is not enabled. [DEFAULT]
+// - "1": Gemm FastMath mode is enabled.
+static const char* const kOrtSessionOptionsGemmFastMathMode = "session.enable_gemm_fastmath_mode";
```

Comment:
This is confusing. What is fastmath? 
First, complier also has a fast math mode. Whether it was enabled or not, at runtime we cannot change it.
Second, we have multiple gemm implementations. Most of them do not support this switch. If we add an option here but it has no effect, users may report it as a bug.

Reply:
Hi @snnn ,
(1) fastmath mode will enable SBGemm kernel which takes fp32 weigths, activations and perform gemm in Bfloat16 precision to accelerate. I have used the similar terminology as in Arm compute Library [ACL](https://github.com/ARM-software/ComputeLibrary).
(2) yes, when disabled,these kernels are not even invoked. the regular SGemm kernels get used and produce the same result for the same input
(3) this is not the compiler fast math mode. this can be enabled by setting the session options at runtime. I have added test cases for this,
   ```
 SessionOptions so;
    ASSERT_STATUS_OK(so.config_options.AddConfigEntry(
        kOrtSessionOptionsGemmFastMathMode, "1"));
```
(4) In the commit and the PR I stated it is only for aarch64 and sgemm acceleration with sbgemm only. But please let me know if I need to rename the flag or update any doc to state the scope explicitly.